js/imageslider/shared/SliderPreview.svelte

```diff
@@ -0,0 +1,192 @@
+	import { DownloadLink } from "@gradio/wasm/svelte";
+	import { ZoomableImage } from "./zoom";
+	import { onMount } from "svelte";
+	import { tweened, type Tweened } from "svelte/motion";
+	export let value: [null | FileData, null | FileData] = [null, null];
+	export let label: string | undefined = undefined;
+	export let show_download_button = true;
+	export let show_label: boolean;
+	export let i18n: I18nFormatter;
+	export let position: number;
+	export let layer_images = true;
+	export let show_single = false;
+	export let slider_color: string;
+	export let show_fullscreen_button = true;
+	export let el_width = 0;
+	export let max_height: number;
+
+	let img: HTMLImageElement;
+	let slider_wrap: HTMLDivElement;
+	let image_container: HTMLDivElement;
+
+	let transform: Tweened<{ x: number; y: number; z: number }> = tweened(
+		{ x: 0, y: 0, z: 1 },
+		{
+			duration: 75
+		}
+	);
+	let parent_el: HTMLDivElement;
+
+	$: coords_at_viewport = get_coords_at_viewport(
+		position,
+		viewport_width,
+		$transform.x,
+		$transform.z
+	);
+	$: style = layer_images
+		? `clip-path: inset(0 0 0 ${coords_at_viewport * 100}%)`
+		: "";
+
+	function get_coords_at_viewport(
+		viewport_percent_x: number, // 0-1
+		viewportWidth: number,
+		tx: number, // image translation x (in pixels)
+
+		scale: number // image scale (uniform)
+	): number {
+		const vx = viewport_percent_x * viewportWidth;
+
+		const ix = (vx - tx) / scale / img_width;
+
+		return ix;
+	}
+
+	let img_width = 0;
+	let viewport_width = 0;
+
+	let zoomable_image: ZoomableImage | null = null;
+	let observer: ResizeObserver | null = null;
+
+	function init_image(
+		img: HTMLImageElement,
+		slider_wrap: HTMLDivElement
+	): void {
+		if (!img || !slider_wrap) return;
+		zoomable_image?.destroy();
+		observer?.disconnect();
+		img_width = img?.getBoundingClientRect().width || 0;
+		viewport_width = slider_wrap?.getBoundingClientRect().width || 0;
+		zoomable_image = new ZoomableImage(slider_wrap, img);
+		zoomable_image.subscribe(({ x, y, scale }) => {
+			transform.set({ x, y, z: scale });
+		});
+
+		observer = new ResizeObserver((entries) => {
+			for (const entry of entries) {
+				if (entry.target === slider_wrap) {
+					viewport_width = entry.contentRect.width;
+				}
+
+				if (entry.target === img) {
+					img_width = entry.contentRect.width;
+				}
+			}
+		});
+		observer.observe(slider_wrap);
+		observer.observe(img);
+	}
+
+	$: init_image(img, slider_wrap);
+
+	onMount(() => {
+		return () => {
+			zoomable_image?.destroy();
+			observer?.disconnect();
+		};
+	});
+
+	let is_full_screen = false;
+	let slider_wrap_parent: HTMLDivElement;
+</script>
+
+<BlockLabel {show_label} Icon={Image} label={label || i18n("image.image")} />
+{#if (value === null || value[0] === null || value[1] === null) && !show_single}
+	<Empty unpadded_box={true} size="large"><Image /></Empty>
+{:else}
+	<div class="image-container" bind:this={image_container}>
+		<IconButtonWrapper>
```

Comment:
I think we should add a clear button similar to the image editor

Reply:
Yeah I agree. It actually has one but it doesn't get shown because of how I'm handling the 'preview' after uploading. Will fix.

Reply:
done!